CollapseLauncher/Classes/Helper/WindowUtility.cs

```diff
@@ -592,6 +595,32 @@ private static IntPtr MainWndProc(IntPtr hwnd, uint msg, UIntPtr wParam, IntPtr
 
                         break;
                     }
+                case WM_QUERYENDSESSION:
+                    // Return FALSE (0) to prevent shutdown if critical operation is in progress
+                    if (MainWindow.IsCriticalOpInProgress)
+                    {
+                        // Display reason using ShutdownBlocker
+                        ShutdownBlocker.StartBlocking(CurrentWindowPtr, Locale.Lang._Dialogs.EnsureExitSubtitle,
+                                                      ILoggerHelper.GetILogger("ShutdownBlocker"));
```

Comment:
Another lines that can be removed. As the StartBlocking should've been set on ``MainWindow.IsCriticalOpInProgress``